src/components/ScaleLattice.vue

```diff
@@ -0,0 +1,340 @@
+    return props.scale.intervals.reduce(
+      (canLattice: boolean, interval: Interval) =>
+        canLattice &&
+        interval.monzo
+          .toIntegerMonzo()
+          .slice(9)
+          .reduce(
+            (isZero: boolean, component: number) => isZero && component === 0,
+            true
+          ),
+      true
+    );
+  } catch {
+    return false;
+  }
+});
+
+function maybeGetPrimeEquave(equave: Interval): number | null {
+  try {
+    const intergerMonzo: number[] = equave.monzo.toIntegerMonzo();
+    return intergerMonzo.reduce(
+      (
+        primeIndex: number | null,
+        component: number,
+        i: number
+      ): number | null => {
+        if (component === 1 && !primeIndex) {
+          return i;
+        }
+        if (component !== 0) {
+          console.error("Equave is not a prime number");
+          return primeIndex;
+        }
+        if (component === 0) {
+          return primeIndex;
+        }
+        return primeIndex;
+      },
+      null
+    );
+  } catch {
+    return null;
+  }
+}
+
+const nodes: ComputedRef<Node[] | string> = computed(() => {
+  try {
+    const equavePrimeIndex = maybeGetPrimeEquave(props.scale.equave);
+    return props.scale.intervals.map(({ monzo, name: label }, index) => {
+      const node: Node = monzo.toIntegerMonzo().reduce(
+        (node, component, i): Node => {
+          const { x, y } = node;
+          if (i === equavePrimeIndex) {
+            return node;
+          }
+
+          const componentVal = component.valueOf();
+          if (componentVal === 0) {
+            return node;
+          } else {
+            const p: string = PRIMES[i].toString();
+            const vector = coords[p] || { x: 0, y: 0 };
+            return {
+              ...node,
+              x: x + vector.x * componentVal,
+              y: y + vector.y * componentVal,
+            };
+          }
+        },
+        { x: 0, y: 0, label, index, fill: "white" }
+      );
+      return node;
+    });
+  } catch {
+    return "Cannot make lattice of non JI scale.";
+  }
+});
+
+const nodesByLabel: ComputedRef<{ [key: string]: Node }> = computed(() =>
+  Array.isArray(nodes.value)
+    ? nodes.value.reduce((acc, node) => ({ ...acc, [node.label]: node }), {})
+    : {}
+);
+
+const edges = computed(() => {
+  try {
+    if (!Array.isArray(nodes.value)) {
+      return [];
+    }
+    const combinations = getPairCombinations(props.scale.intervals);
+    return combinations
+      .map(([a, b]) => {
+        return {
+          distance: monzoEuclideanDistance(
+            a.monzo.toIntegerMonzo().slice(1),
+            b.monzo.toIntegerMonzo().slice(1)
+          ),
+          pair: [a.name, b.name],
+        };
+      })
+      .filter(({ distance }) => distance === 1);
+  } catch {
+    return [];
+  }
+});
+
+const svgElement = ref<HTMLElement | null>(null);
+
+const viewBox = computed(() => {
+  const defaultViewBox = { minX: 0, minY: 0, maxX: 0, maxY: 0 };
```

Comment:
It's better to start with minima set to `Infinity` and maxima set to `-Infinity` before reducing.